rpcsx-gpu/CMakeLists.txt

```diff
@@ -8,5 +8,9 @@ add_executable(rpcsx-gpu
 target_include_directories(rpcsx-gpu PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 target_link_libraries(rpcsx-gpu PUBLIC amdgpu::bridge amdgpu::device glfw Vulkan::Vulkan rx)
 set_target_properties(rpcsx-gpu PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
-target_link_options(rpcsx-os PUBLIC "LINKER:-Ttext-segment,0x0000060000000000")
+if(CMAKE_CXX_COMPILER_LINKER_ID MATCHES "^(LLD|GNU|MOLD)$")
```

Comment:
GNU is GNU ld, it do not support --image-base
Also, seems better to create cmake function to specify base, because we have 2 places with such flags that should be in sync

Reply:
There are [two GNU ld versions](https://en.wikipedia.org/wiki/Linker_(computing)#GNU_linker) and only `ld.gold` does no support `--image-base`. According to [CMake documentation](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_LINKER_ID.html) `GNU` is for `ld.bfd` and `GNUgold` is for `ld.gold`.

Reply:
```bash
$ ld.bfd --version
GNU ld (GNU Binutils for Ubuntu) 2.43.1
Copyright (C) 2024 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or (at your option) a later version.
This program has absolutely no warranty.
```

```bash
$ ld.bfd --image-base=0x0000060000000000
ld.bfd: unrecognized option '--image-base=0x0000060000000000'
ld.bfd: use the --help option for usage information
```

```bash
$ ld.gold --version
GNU gold (GNU Binutils for Ubuntu 2.43.1) 1.16
Copyright (C) 2024 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or (at your option) a later version.
This program has absolutely no warranty.
```

```bash
$ ld.gold --image-base=0x0000060000000000
ld.gold: --image-base=0x0000060000000000: unknown option
ld.gold: use the --help option for usage information
```

With your changes linking with any GNU linker is broken, did you test them?